Data/Primitive/Array.hs

```diff
@@ -341,6 +343,11 @@ instance Eq a => Eq (Array a) where
    where loop i | i < 0     = True
                 | otherwise = indexArray a1 i == indexArray a2 i && loop (i-1)
 
+instance Eq1 Array where
+  liftEq p a1 a2 = sizeofArray a1 == sizeofArray a2 && loop (sizeofArray a1 - 1)
+   where loop i | i < 0     = True
+                | otherwise = p (indexArray a1 i) (indexArray a2 i) && loop (i-1)
+
```

Comment:
I just merged my eagerness patch, so you should use `indexArray##` here. Also, if we're adding `Eq1` we should surely also add `Ord1`.